### DISCUSSION TITLE
Top, Bottom. Left. right won't work for non zero values.

### DISCUSSION THREAD

**Discussion by shivangrathore** [tailwindlabs/tailwindcss#12108] [Help] [answered]
I am not sure if its a bug or a user error. But I am running into wierd issue.
When I am doing `[top | bottom | left | right]-full`, even `translate-[]-full`, it doesn't apply any css at all.
I have already tested enough but can't find the issue. they only work of zero values.

![image](https://github.com/tailwindlabs/tailwindcss/assets/61678021/561b515d-e00c-4038-96e3-e1aea56ebccd)
![image](https://github.com/tailwindlabs/tailwindcss/assets/61678021/6bfeb094-3a83-4d27-8131-6eea097a1274)

While after adding those classes in safeList config it works as expected. What I am missing.

P.S. I am using `transform` with `-translate` classes

**Comment by jayantghadge** [comment]
If it works well in 'safeList' then try double checking your 'tailwind.config.js' file for purge configuration. In Tailwind V3, purge has been renamed as 'content'.

**Comment by alex-krasikau** [comment]
Try reproducing it on https://play.tailwindcss.com/ and we can help.

**Comment by shivangrathore** [comment, marked as answer]
> Try reproducing it on https://play.tailwindcss.com/ and we can help. 


Can't reproduce on playtailwind, I am not sure why this was happening but I found a solution, I was defining my alert in same file as my AlertContextProvider (I am using nextjs)

When I tried to reproduce 2-3 times it clicked in my mind that it works when I move Alert Component 
![image](https://github.com/tailwindlabs/tailwindcss/assets/61678021/91b577fe-e1c1-4acf-aa33-fd843f58771a)
outside of that alert-context file, it magically worked.


For some reason tailwind wasn't able to scan the classnames used in alert-context file. Although thanks for willing to help ❤️
